flock_aix.go

```diff
@@ -0,0 +1,265 @@
+// Copyright 2019 Tim Heckman. All rights reserved. Use of this source code is
+// governed by the BSD 3-Clause license that can be found in the LICENSE file.
+
+// This code implements the filelock API using POSIX 'fcntl' locks, which attach
+// to an (inode, process) pair rather than a file descriptor. To avoid unlocking
+// files prematurely when the same file is opened through different descriptors,
+// we allow only one read-lock at a time.
+//
+// This code is adapted from the Go package:
+// cmd/go/internal/lockedfile/internal/filelock
```

Comment:
The portions of this implementation taken from the Go project should carry the Go license and copyright notice.

(It is of course fine to combine that code with other compatibly-licensed code, but it is important to comply with the terms of the license.)